runtime/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttClientFactory.java

```diff
@@ -2282,14 +2284,21 @@ private void doEncodeConnect(
             final int propertiesSize0 = propertiesSize;
             final int willSize = will != null ? will.sizeof() : 0;
             flags |= will != null ? (WILL_FLAG_MASK | ((willMessage.flags() & RETAIN_MASK) != 0 ? WILL_RETAIN_MASK : 0)) : 0;
-            final MqttConnectV5FW connect =
-                mqttConnectV5RW.wrap(writeBuffer, FIELD_OFFSET_PAYLOAD, writeBuffer.capacity())
+
+            final MqttConnectHeaderFW connectHeader =
+                mqttConnectHeaderRW.wrap(writeBuffer, FIELD_OFFSET_PAYLOAD, writeBuffer.capacity())
                     .typeAndFlags(0x10)
                     .remainingLength(11 + propertiesSize0 + clientId.length() + 2 + willSize)
                     .protocolName(MQTT_PROTOCOL_NAME)
                     .protocolVersion(MQTT_PROTOCOL_VERSION)
                     .flags(flags)
                     .keepAlive((int) MILLISECONDS.toSeconds(keepAliveMillis))
+                    .build();
+
+            doNetworkData(traceId, authorization, 0L, connectHeader);
+
+            final MqttConnectV5FW connect =
+                mqttConnectV5RW.wrap(writeBuffer, FIELD_OFFSET_PAYLOAD, writeBuffer.capacity())
```

Comment:
Let's send the combined connect header plus properties plus client-id in a single call to `doNetworkData(...)` passing the aggregated write buffer, from beginning of connect header to end of connect.